woocommerce/woocommerce-android
# Issue/track order creation ipp flows

### Description
[This PR mirrors the changes introduced in this PR on iOS](https://github.com/woocommerce/woocommerce-ios/pull/8764).

I shamelessly copy-pasted the description but also parts of the code, thanks @toupper!

==============================

We track here the duration of Order Creation and IPP flows to gather good baseline numbers to evaluate the improvements of the Speedy IPP projects.

We measure:

Flow 1: Add new order button tapped -> order was successfully created.
Flow 2: Add new order button tapped -> card payment method was selected.
Flow 3: Add new order button tapped -> card payment collection was successful (in both regular and interac).
Flow 4: Card payment method was selected -> Card payment collection was successful (in both regular and interac).

We reported those values by adding a new properties to the related events:

| Event      | New properties |
| ----------- | ----------- |
| `"order_creation_success" `     | `"milliseconds_since_order_add_new"`       |
| `"payments_flow_collect"`   | `"milliseconds_since_order_add_new"`        |
|`"card_present_collect_payment_success"` |  `"milliseconds_since_order_add_new"`, `"milliseconds_since_card_collect_payment_flow"`|


P.S.  This is a naive solution which doesn't take into account scenarios in which the user for example creates an order and collects a payment for a different order later, etc. However, we don't need a fully precise data and this solution should give us a good baseline we can use to measure against the improvements we plan to introduce in the upcoming months.


## Testing instructions


1. Go to Orders
2. Tap on New
3. Add products to the order, and tap on Create
4. Tap on Collect payment
5. Follow the flow to collect the payment with card
6. Check the logs contain the attributes with duration in ms.

Check that the events are tracked, e.g.:
`"order_creation_success" `:
`🔵 Tracked: order_creation_success, Properties: {"milliseconds_since_order_add_new":"8244","blog_id":154993204,"is_wpcom_store":false,"is_debug":true}`

`"payments_flow_collect"`:
`🔵 Tracked: payments_flow_collect, Properties: {"payment_method":"card","flow":"order_payment","milliseconds_since_order_add_new":"14521","blog_id":154993204,"is_wpcom_store":false,"is_debug":true}`

`card_present_collect_payment_success`:

US: `🔵 Tracked: card_present_collect_payment_success, Properties: {"milliseconds_since_order_add_new":"42795","milliseconds_since_card_collect_payment_flow":"28275","plugin_slug":"woocommerce-payments","country":"US","currency":"USD","payment_method_type":"card","card_reader_model":"STRIPE_M2","card_reader_battery_level":"17 %","blog_id":154993204,"is_wpcom_store":false,"is_debug":true}`

CA: `🔵 Tracked: card_present_collect_interac_payment_success, Properties: {"milliseconds_since_order_add_new":"39505","milliseconds_since_card_collect_payment_flow":"28043","plugin_slug":"woocommerce-payments","country":"CA","currency":"CAD","payment_method_type":"card_interac","card_reader_model":"WISEPAD_3","card_reader_battery_level":"63 %","blog_id":206146642,"is_wpcom_store":true,"is_debug":true}`

Repeat the process to check that the measurements are properly resetted after a successful order.

==============================
